<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>831</width>
    <height>474</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>DuSE-MT</string>
  </property>
  <property name="windowIcon">
   <iconset resource="duse-mt.qrc">
    <normaloff>:/icons/duse-mt-32x32.png</normaloff>:/icons/duse-mt-32x32.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidgetLayout">
   <layout class="QGridLayout" name="gridLayout_3">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>3</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QTabWidget" name="centralWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabWelcome">
       <attribute name="title">
        <string>Welcome to DuSE-MT</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_13">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QWidget" name="welcomeViewWidget" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabModelView">
       <attribute name="title">
        <string>Input Model</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_11">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QWidget" name="modelViewWidget" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabDesignSpaceLocationView">
       <attribute name="title">
        <string>Current Design Space Location</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_15">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QWidget" name="designSpaceLocationViewWidget" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabParetoFront">
       <attribute name="title">
        <string>Pareto Front</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_12">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="QWidget" name="paretoFrontViewWidget" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>831</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionFileNewModel"/>
    <addaction name="actionFileOpenModel"/>
    <addaction name="separator"/>
    <addaction name="actionFileNewDuseDesign"/>
    <addaction name="actionFileOpenDuseDesign"/>
    <addaction name="separator"/>
    <addaction name="actionFileSave"/>
    <addaction name="actionFileSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionFileQuit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionHelpAboutDuSEMT"/>
    <addaction name="actionHelpAboutPlugins"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_Window">
    <property name="title">
     <string>&amp;Window</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_Actions">
    <property name="title">
     <string>&amp;Actions</string>
    </property>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
   <addaction name="menu_Actions"/>
   <addaction name="menu_Window"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="windowTitle">
    <string>Main Toolbar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionFileNewModel"/>
   <addaction name="actionFileOpenModel"/>
   <addaction name="actionFileSave"/>
   <addaction name="separator"/>
   <addaction name="actionFileQuit"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dckIssues">
   <property name="windowTitle">
    <string>Issues</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QGridLayout" name="gridLayout_4">
     <property name="leftMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <widget class="QListView" name="txeIssues">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dckXPath">
   <property name="windowTitle">
    <string>XPath Browser</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_5">
     <property name="leftMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <widget class="QTextEdit" name="txeXPath"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dckOcl">
   <property name="windowTitle">
    <string>OCL Browser</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout_6">
     <property name="leftMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <widget class="QTextEdit" name="txeOcl"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dckJavaScript">
   <property name="windowTitle">
    <string>JavaScript Browser</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QGridLayout" name="gridLayout_7">
     <property name="leftMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>JavaScript Editor</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_8">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="1" column="0">
           <widget class="QPushButton" name="psbJSEvaluate">
            <property name="text">
             <string>&amp;Evaluate</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QTextEdit" name="txeJavaScript"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>JavaScript Output</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_9">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTextEdit" name="txeJavaScriptEvaluation">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dckMetrics">
   <property name="minimumSize">
    <size>
     <width>180</width>
     <height>42</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Quality Metrics</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_5">
    <layout class="QGridLayout" name="gridLayout_10">
     <property name="leftMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <widget class="QWidget" name="metricsLayout" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dckDesignSpace">
   <property name="windowTitle">
    <string>Design Space</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_6">
    <layout class="QGridLayout" name="gridLayout_14">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <property name="verticalSpacing">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <widget class="QTableWidget" name="tblDesignSpace">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <column>
        <property name="text">
         <string>Design Dimension</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Dimension Instance Target</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Variation Point</string>
        </property>
       </column>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="btnOptimize">
       <property name="text">
        <string>&amp;Optimize</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionFileNewModel">
   <property name="icon">
    <iconset theme="document-new" resource="duse-mt.qrc">
     <normaloff>:/icons/document-new.png</normaloff>:/icons/document-new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;New Model...</string>
   </property>
   <property name="toolTip">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionFileOpenModel">
   <property name="icon">
    <iconset theme="document-open" resource="duse-mt.qrc">
     <normaloff>:/icons/document-open.png</normaloff>:/icons/document-open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open Model ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionFileSave">
   <property name="icon">
    <iconset theme="document-save" resource="duse-mt.qrc">
     <normaloff>:/icons/document-save.png</normaloff>:/icons/document-save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="toolTip">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionFileSaveAs">
   <property name="icon">
    <iconset theme="document-save-as" resource="duse-mt.qrc">
     <normaloff>:/icons/document-save-as.png</normaloff>:/icons/document-save-as.png</iconset>
   </property>
   <property name="text">
    <string>Save &amp;as...</string>
   </property>
  </action>
  <action name="actionFileQuit">
   <property name="icon">
    <iconset theme="application-exit" resource="duse-mt.qrc">
     <normaloff>:/icons/application-exit.png</normaloff>:/icons/application-exit.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionHelpAboutPlugins">
   <property name="icon">
    <iconset theme="help-about" resource="duse-mt.qrc">
     <normaloff>:/icons/help-about.png</normaloff>:/icons/help-about.png</iconset>
   </property>
   <property name="text">
    <string>About &amp;Plugins...</string>
   </property>
  </action>
  <action name="actionHelpAboutDuSEMT">
   <property name="icon">
    <iconset resource="duse-mt.qrc">
     <normaloff>:/icons/duse-mt-22x22.png</normaloff>:/icons/duse-mt-22x22.png</iconset>
   </property>
   <property name="text">
    <string>About &amp;DuSE-MT</string>
   </property>
   <property name="toolTip">
    <string>About DuSE-MT</string>
   </property>
  </action>
  <action name="actionFileNewDuseDesign">
   <property name="icon">
    <iconset theme="document-new">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>New &amp;DuSE Design ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="actionFileOpenDuseDesign">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Open DuSE Design ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+U</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="duse-mt.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionFileQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>415</x>
     <y>236</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
